src/main/java/me/voidxwalker/autoreset/api/seedprovider/SeedProvider.java

```diff
@@ -17,17 +20,16 @@ default boolean shouldShowSeed() {
     }
 
     /**
-     * Waits for a seed to be available. This should only be called by Atum while using SeedQueue.
+     * Gets the waiting screen. The implemented screen only needs to override the render method. It may also allow for
+     * cancelling of seeds with esc, or it can call cancelWorldCreation through other means such as a button.
      */
-    default void waitForSeed() {
+    default Optional<AtumWaitingScreen> getWaitingScreen() {
+        return Optional.empty();
     }
 
     /**
-     * Gets the waiting screen.
-     * The implemented waiting screen should run the provided continueWorldCreation method once a seed is available, or alternatively cancelWorldCreation.
-     * The implemented waiting screen can also override shouldCloseOnEsc(), returning true to allow cancelling with the 'escape' key.
+     * Runs when an exception or cancellation occurs while resolving the seed future.
```

Comment:
Maybe you could mention that there is no guarantees of when this gets called or how often and that it always happens on the client thread